src/vulkan/engine_vulkan.cc

```diff
@@ -106,20 +117,26 @@ Result EngineVulkan::Shutdown() {
 Result EngineVulkan::AddRequirement(Feature feature,
                                     const Format* fmt,
                                     uint32_t val) {
-  auto it = std::find_if(requirements_.begin(), requirements_.end(),
-                         [&feature](const EngineVulkan::Requirement& req) {
-                           return req.feature == feature;
-                         });
-  if (it != requirements_.end())
-    return Result("Vulkan::Feature Already Exists");
-
   if (feature == Feature::kFenceTimeout) {
     fence_timeout_ms_ = val;
     return {};
   }
 
-  requirements_.push_back({feature, fmt});
-  return {};
+  if (feature == Feature::kFramebuffer) {
+    if (fmt != nullptr)
+      color_frame_format_ = MakeUnique<Format>(*fmt);
```

Comment:
You can't MakeUnique here, you don't know who owns fmt, so you can't just take ownership.

Reply:
Why not just set this to the ToVkFormat value ?